wordle-web/pages/wordList.vue

```diff
@@ -0,0 +1,280 @@
+          <td>
+            <v-tooltip text="mark the word as a common word">
+              <template v-slot:activator="{ props }">
+                <v-btn
+                  :disabled="!signedIn"
+                  color="primary"
+                  v-bind="props"
+                  @click="markAsCommon(words.word, words.isCommon), refreshWords()"
+                  >Common</v-btn
+                >
+              </template>
+            </v-tooltip>
+          </td>
+          <td>
+            <v-tooltip text="delete the word">
+              <template v-slot:activator="{ props }">
+                <v-btn
+                  :disabled="!canAddWords"
+                  color="primary"
+                  v-bind="props"
+                  @click="confirmDeleteWord(words.word), refreshWords()"
+                  >Delete</v-btn
+                >
+              </template>
+            </v-tooltip>
+          </td>
+        </tr>
+      </tbody>
+    </v-table>
+    <v-pagination
+      v-model="pageNumber"
+      :length="Math.ceil(totalCount / pageSize)"
+      @input="getWordList(wordToSearch, pageNumber, pageSize), refreshWords()"
+      variant="outlined"
+    />
+    <!---->
+
+    <!---->
+    <div class="d-flex">
+      <v-text-field
+        clearable
+        label="Add A Word"
+        v-model="wordToAdd"
+      ></v-text-field>
+      <v-tooltip text="Add a word to the word list">
+        <template v-slot:activator="{ props }">
+          <v-btn
+            :disabled="!canAddWords"
+            color="primary"
+            v-bind="props"
+            @click="addWord(), refreshWords()"
+            >Add Word</v-btn
+          >
+        </template>
+      </v-tooltip>
+    </div>
+  </v-card>
+</template>
+<script setup lang="ts">
+import TokenService from "~/scripts/tokenService";
+import Axios from "axios";
+const showSignInDialog = ref(false);
+const tokenService = new TokenService();
+const wordToAdd = ref("");
+const wordToSearch = ref("");
+const wordToDelete = ref("");
+const wordList = ref<WordListItems[]>();
+const totalCount = ref(0);
+const pageNumber = ref(1);
+const pageSize = ref(10);
+const showConfirmDeleteWord = ref(false);
+const userNameGlobal: Ref<string> = inject("userName")! as Ref<string>;
+provide("wordToDelete", wordToDelete);
+const signedIn = computed(() => tokenService.isLoggedIn());
+const parseToken = computed(() => tokenService.parseToken());
+const canAddWords = computed(() => {
+  if (
+    parseToken.value[5] == "true" &&
+    parseInt(parseToken.value[4]) >= 21 &&
+    signedIn.value == true
+  ) {
+    return true;
+  } else {
+    return false;
+  }
+});
+
+interface WordListItems {
+  word: string;
+  isCommon: boolean;
+}
+
+onMounted(async () => {
+  if (tokenService.isLoggedIn()) {
+    showSignInDialog.value = false;
+  }
+  await refreshWords();
+});
+
+watch([wordToSearch, pageNumber, pageSize], async () => {
+  await refreshWords();
+});
+
+async function refreshWords(){
```

Comment:
I'm not sure if this needs to be called for every update. You could store some information on the front end and update it on the frontend and just update it through the post endpoints you've implemented. But I wasn't the one to implement autocomplete on my team's assignment, so this might be required for that.

Reply:
We'll look into this, thanks for this suggestion!